src/OrleansAzureUtils/Providers/Storage/AzureBlobStorage.cs

```diff
@@ -0,0 +1,299 @@
+    /// <c>ContainerName</c> -- defaults to <c>grainstate</c>
+    /// <c>SerializeTypeNames</c> -- defaults to <c>OrleansGrainState</c>
+    /// <c>PreserveReferencesHandling</c> -- defaults to <c>false</c>
+    /// <c>UseFullAssemblyNames</c> -- defaults to <c>false</c>
+    /// <c>IndentJSON</c> -- defaults to <c>false</c>
+    /// </para>
+    /// </remarks>
+    /// <example>
+    /// Example configuration for this storage provider in OrleansConfiguration.xml file:
+    /// <code>
+    /// &lt;OrleansConfiguration xmlns="urn:orleans">
+    ///   &lt;Globals>
+    ///     &lt;StorageProviders>
+    ///       &lt;Provider Type="Orleans.Storage.AzureBlobStorage" Name="AzureStore"
+    ///         DataConnectionString="UseDevelopmentStorage=true"
+    ///       />
+    ///   &lt;/StorageProviders>
+    /// </code>
+    /// </example>
+    public class AzureBlobStorage : IStorageProvider
+    {
+        private JsonSerializerSettings settings;
+
+        private CloudBlobContainer container;
+
+        /// <summary> Logger used by this storage provider instance. </summary>
+        /// <see cref="IStorageProvider.Log"/>
+        public Logger Log { get; private set; }
+
+        /// <summary> Name of this storage provider instance. </summary>
+        /// <see cref="IProvider.Name"/>
+        public string Name { get; private set; }
+
+        /// <summary> Initialization function for this storage provider. </summary>
+        /// <see cref="IProvider.Init"/>
+        public async Task Init(string name, IProviderRuntime providerRuntime, IProviderConfiguration config)
+        {
+            Log = providerRuntime.GetLogger("Storage.AzureBlobStorage");
+
+            try
+            {
+                this.Name = name;
+                ConfigureJsonSerializerSettings(config);
+
+                if (!config.Properties.ContainsKey("DataConnectionString")) throw new BadProviderConfigException("The DataConnectionString setting has not been configured in the cloud role. Please add a DataConnectionString setting with a valid Azure Storage connection string.");
+
+                var account = CloudStorageAccount.Parse(config.Properties["DataConnectionString"]);
+                var blobClient = account.CreateCloudBlobClient();
+                var containerName = config.Properties.ContainsKey("ContainerName") ? config.Properties["ContainerName"] : "grainstate";
+                container = blobClient.GetContainerReference(containerName);
+                await container.CreateIfNotExistsAsync().ConfigureAwait(false);
+
+                Log.Info((int)AzureProviderErrorCode.AzureBlobProvider_InitProvider, "Init: Name={0} ServiceId={1} {2}", name, providerRuntime.ServiceId.ToString(), string.Join(" ", FormatPropertyMessage(config)));
+                Log.Info((int)AzureProviderErrorCode.AzureBlobProvider_ParamConnectionString, "AzureBlobStorage Provider is using DataConnectionString: {0}", config.Properties["DataConnectionString"]);
```

Comment:
hmmm, is the other provider doing the same thing? I'm extremely hesitant of logging credentials, as it's very easy to end up in logs that many can access. Orleans typically scrubs the account key in places where it does log it. See `ConfigUtilities.RedactConnectionStringInfo`